script/management_graphs.Rmd

```diff
@@ -0,0 +1,209 @@
+---
+title: "Management_graphs"
+author: "Sander Devisscher"
+date: '2022-08-17'
+output:
+  word_document: default
+  html_document: default
+---
+
+```{r setup, include=FALSE}
+library(rgbif)
+library(geojsonsf)
+library(sf)
+library(tidyverse)
+```
+
+```{r, include=FALSE}
+cleanData <- read_csv("./interim/Lithobates_catesbeianus_management_data.csv")
+```
+
+# GRAPH: Total number of bullfrogs caught
+```{r total individual caught, include=FALSE}
+total_by_year <- cleanData %>% 
+  group_by(year) %>% 
+  summarise(n = sum(individualCount, na.rm = TRUE)) %>% 
+  ggplot(aes(x = year, y = n)) +
+  geom_bar(stat = "identity")
+```
+
+```{r echo=FALSE}
+plot(total_by_year)
+```
+
+# GRAPH: Absolute number of bullfrogs caught by lifestage
+```{r total individuals caught by lifestage, include=FALSE}
+total_by_year_by_lifestage <- cleanData %>% 
+  group_by(year, lifeStage) %>% 
+  summarise(n = sum(individualCount, na.rm = TRUE)) %>% 
+  filter(!is.na(lifeStage)) %>% 
+  ggplot(aes(x = year, y = n, fill = lifeStage)) +
+  geom_bar(stat = "identity",
+           position = "stack")
+```
+
+```{r echo=FALSE}
+total_by_year_by_lifestage
+```
+
+# GRAPH: Cumulative number of bullfrogs caught
+```{r include=FALSE}
+cum_by_year <- cleanData %>% 
+  arrange(year) %>% 
+  mutate(cum = cumsum(individualCount)) %>% 
+  group_by(year) %>% 
+  summarise(n = max(cum)) %>% 
+  ggplot(aes(x = year, y = n)) +
+  geom_bar(stat = "identity")
+```
+
+```{r echo=FALSE}
+plot(cum_by_year)
+```
+
+
+# GRAPH: Catch per unit of effort
+```{r total individuals caught by unit of effort, include=FALSE}
+total_by_year_corrected <- cleanData %>% 
+  group_by(year) %>% 
+  summarise(effort = n_distinct(eventID),
+            n = sum(individualCount, na.rm = TRUE)/effort) 
+
+plot_cpue <- total_by_year_corrected %>% 
+  ggplot(aes(x = year, y = n)) +
+  geom_bar(stat = "identity") + 
+  geom_line(aes(x = year, y = effort),
+            stat = "identity") +
+  scale_y_continuous(name = "Total catch per unit of effort",
+                     sec.axis = sec_axis(trans=~., name="effort"))
+```
+
+```{r echo=FALSE}
+plot(plot_cpue)
+```
+
+# GRAPH: Catch per unit of effort per lifstage
+
+```{r capture per unit of effort by lifestage, include=FALSE}
+total_by_year_corrected_lifestage <- cleanData %>% 
+  group_by(year, lifeStage) %>% 
+  mutate(effort = n_distinct(eventID),
+         n = as.integer(sum(individualCount, na.rm = TRUE))) %>% 
+  group_by(year, lifeStage) %>% 
+  summarise(effort = max(effort),
+            n = sum(n, na.rm = TRUE)/effort) %>% 
+  ungroup() %>% 
+  add_row(year = 2013,
```

Comment:
is bedoeling hier een ontbrekende waarde op te lossen? Merk op dat functie 'expand' ook gebruikt kan worden (robuuster voor gelijkaardige situaties in de toekomst): https://www.rdocumentation.org/packages/tidyr/versions/0.8.3/topics/expand

Reply:
goed punt